src/tests/app.test.ts

```diff
@@ -1,32 +1,75 @@
 import 'jest';
 import supertest from 'supertest';
+import logger from '../logger';
 
-/* eslint-disable @typescript-eslint/no-var-requires, global-require */
+const loggerErrorSpy = jest.spyOn(logger, 'error').mockImplementation();
+const loggerInfoSpy = jest.spyOn(logger, 'info').mockImplementation();
+
+// eslint-disable-next-line import/first
+import { app, initSlack } from '../app';
+
+const processExitSpy = jest.spyOn(process, 'exit').mockImplementation();
+const mockSlackAuth = jest.fn();
+jest.mock('@slack/web-api', () => ({
+  // eslint-disable-next-line object-shorthand, func-names, @typescript-eslint/explicit-function-return-type, space-before-function-paren
+  WebClient: function() {
+    return {
+      auth: {
+        test: mockSlackAuth,
+      },
+    };
+  },
+}));
 
 describe('app', () => {
   beforeEach(() => {
-    process.env.SLACK_SIGNING_SECRET = 'A JUNK TOKEN';
-    process.env.SLACK_BOT_TOKEN = 'ANOTHER JUNK TOKEN';
+    Object.defineProperty(process.env, 'NODE_ENV', { value: 'test' });
+    jest.resetModules();
+    jest.resetAllMocks();
+    mockSlackAuth.mockRejectedValue('Invalid Auth');
   });
 
   it('responds successfully to GET /', (done) => {
-    const { app } = require('../app');
     supertest(app)
       .get('/')
       .expect(200, done);
   });
 
-  it("fails to start when Slack tokens are not provided and env isn't test", () => {
-    delete process.env.SLACK_BOT_TOKEN;
+  it("fails to initialize Slack when Slack tokens are not provided and env isn't test and exits", async () => {
```

Comment:
Wanted to move this around so that the intent was more clear

```suggestion
  it("will exit the process when Slack tokens are not provided and NODE_ENV !== "test", because it cannot initialize Slack", async () => {
```